src/travel-details-map-screen/TravelDetailsMapScreenComponent.tsx

```diff
@@ -43,35 +46,87 @@ export const TravelDetailsMapScreenComponent = ({
   const {location: geolocation} = useGeolocationState();
 
   const features = useMemo(() => createMapLines(legs), [legs]);
-  const bounds = getMapBounds(features);
+  const bounds = !vehicleWithPosition ? getMapBounds(features) : undefined;
+  const centerPosition = vehicleWithPosition?.location
+    ? [
+        vehicleWithPosition?.location?.longitude,
+        vehicleWithPosition?.location?.latitude,
+      ]
+    : undefined;
 
   const {t} = useTranslation();
   const controlStyles = useControlPositionsStyle();
+
+  const [vehicle, setVehicle] = useState<VehicleWithPosition | undefined>(
+    vehicleWithPosition,
+  );
+  useGetLiveServiceJourneyVehicles(
+    setVehicle,
+    vehicleWithPosition?.serviceJourney?.id,
+  );
+
+  const [shouldTrack, setShouldTrack] = useState<boolean>(true);
+
+  const [followVehicleMapPoint, setFollowVehicleMapPoint] = useState<
+    Coordinates | undefined
+  >();
+
+  useEffect(() => {
+    const location = vehicle?.location;
+
+    if (vehicle === followVehicleMapPoint || !location) return;
```

Comment:
Generally all variables in a `useEffect` should be in the dependency array, and not having it there (or not being able to have it there) might signalise that something else is wrong. Maybe we can have a quick talk about these variables and what they are trying to achieve.